### PR TITLE
parameterize and upate argo v2 images.

### PR DESCRIPTION
**Pull request by HumairAK**
* Add an odh overlay to deploy argo + dspo together, to be utilized by a DSC 
* Parameterize argo images in argo pods 
* update image sources to point to v2 images 
* remove scheduled workflow crd from argo deployment (we already deploy it with dspo) 
* remove priority class from workflow controller because the ODH operator SA does not have permissions to deploy/set this. 
```logs
is forbidden: no PriorityClass with name ds-pipelines-workflow-controller-priorityclass was found
```

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/549/head
git checkout -b pullrequest d0cee33acc6440795ab3a1f424cb55ae80457c96
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-549"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-549`

File: config/argo/kustomization.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ resources:
 - configmap.workflow-controller-configmap.yaml
 - deployment.argo-server.yaml
 # - deployment.workflow-controller.yaml
-- priorityclass.yaml
+# - priorityclass.yaml
```
Comment (gmfrasca) [inline config/argo/kustomization.yaml:20]: we should remove these if they're not needed

File: Makefile [hunk 2]
```diff
@@ -199,6 +199,17 @@ undeploy-kind: ## Undeploy controller from the K8s cluster specified in ~/.kube/
 		&& kustomize edit set namespace ${OPERATOR_NS}
 	kustomize build config/overlays/kind-tests | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
+.PHONY: deployODH
```
[the remark this answers is not in the document]
Comment (HumairAK) [reply Makefile:202]: I think we can make this part of: https://issues.redhat.com/browse/RHOAIENG-1718

File: config/samples/dspa_simple_v2.yaml [hunk 3]
```diff
@@ -4,33 +4,9 @@ metadata:
...
   objectStorage:
     minio:
       deploy: true
       image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
   mlpipelineUI:
-    image: gcr.io/ml-pipeline/frontend:2.0.2
-  workflowController:
-    deploy: true
-    image: gcr.io/ml-pipeline/workflow-controller:v3.3.10-license-compliance
+    image: quay.io/opendatahub/ds-pipelines-frontend:latest
```
[the remark this answers is not in the document]
Comment (HumairAK) [reply config/samples/dspa_simple_v2.yaml:12]: you're right, it does need to be updated

**Pull request merged**